torch_np/_ndarray.py

```diff
@@ -437,7 +437,12 @@ def __setitem__(self, index, value):
 
         if isinstance(value, torch.Tensor):
             value = _util.cast_if_needed(value, self.tensor.dtype)
-        return self.tensor.__setitem__(index, value)
+
+        try:
+            return self.tensor.__setitem__(index, value)
+        except TypeError:
+            value = torch.as_tensor(value, dtype=self.tensor.dtype)
+            return self.tensor.__setitem__(index, value)
```

Comment:
Actually, no, I don't understand this one. Aren't we calling `ndarrays_to_tensors` on values up there? I think that the right fix would:
- Not do anything if `value` is a scalar
- Call `torch.as_tensor` on `value` otherwise.

Reply:
This would introduce the check if value is a scalar for all arguments.

ndarrays_to_tensors is not relevant here, since the failing test has a list value.